src/Slotty.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Wikibase\DataModel;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Adam Shorland
+ * @since 1.9.0
+ */
+interface Slotty {
+
+	/**
+	 * @return array
+	 *     Keys: The name of the slot.
+	 *     Values: The value of the slot.
+	 *         array[]|int[]|string[]|bool[]|float[]
+	 *         A possibly nested structure consisting of only arrays and scalar values
```

Comment:
Well, I guess they could be, but I see no reason they couldn't just be strings.
If we had them as DataValues and then pushed to use this in the API for urls of sitelinks and datatypes or snaks etc that would require a breaking change in the API or more hacks.